src/Compilers/CSharp/Portable/Symbols/TypeWithAnnotations.cs

```diff
@@ -50,13 +51,28 @@ internal struct Builder
             /// </remarks>
             internal bool InterlockedInitialize(TypeWithAnnotations type)
             {
-                if ((object)_defaultType != null)
+                if (!IsDefault)
```

Comment:
Shouldn't this check `_extensions` rather than `_defaultType` if we're setting `_extensions` last?